src/ColorSchemeToggle/index.tsx

```diff
@@ -0,0 +1,64 @@
+import { useContext, type FC, type ReactElement } from "react";
+import ColorContext, { type ColorContextValue } from "../ColorContext";
+import { nextSchemePreference } from "../ColorProvider";
+import { IoMoonSharp, IoSunnySharp } from "react-icons/io5";
+import { AutoIconLight, AutoIconDark } from "./AutoIcon";
+import "./index.scss";
+
+export interface ColorSchemeToggleProps {}
+
+const ColorSchemeToggle: FC<ColorSchemeToggleProps> = ({}) => {
+  let colorSchemeCtx = useContext(ColorContext);
+
+  return (
+    <button
+      className="color-scheme-toggle"
+      aria-label="Toggle light/dark/auto mode"
+      onClick={() => {
+        colorSchemeCtx.setPreference(
+          nextSchemePreference(colorSchemeCtx.preference),
+        );
+      }}
+    >
+      <div
+        data-testid="color-scheme__icon"
+        className={`color-scheme-toggle__icon color-scheme-toggle__icon--${colorSchemeCtx.scheme}`}
+      >
+        {iconForContext(colorSchemeCtx)}
```

Comment:
Now we can make the easy change:

```suggestion
        <Icon />
```